test/builder/QuarkBuilderBorrow.t.sol

```diff
@@ -0,0 +1,877 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+import {QuarkBuilderTest, Accounts, PaymentInfo, QuarkBuilder} from "test/builder/lib/QuarkBuilderTest.sol";
+
+import {Actions} from "src/builder/Actions.sol";
+import {CCTPBridgeActions} from "src/BridgeScripts.sol";
+import {CodeJarHelper} from "src/builder/CodeJarHelper.sol";
+import {CometSupplyMultipleAssetsAndBorrow} from "src/DeFiScripts.sol";
+import {Paycall} from "src/Paycall.sol";
+import {Strings} from "src/builder/Strings.sol";
+
+contract QuarkBuilderBorrowTest is Test, QuarkBuilderTest {
+    uint256 constant BLOCK_TIMESTAMP = 123_456_789;
+    address constant COMET = address(0xc3);
+
+    function borrowIntent_(
+        uint256 amount,
+        string memory assetSymbol,
+        uint256 chainId,
+        uint256[] memory collateralAmounts,
+        string[] memory collateralTokenSymbols
+    ) internal pure returns (QuarkBuilder.BorrowIntent memory) {
+        return QuarkBuilder.BorrowIntent({
+            amount: amount,
+            assetSymbol: assetSymbol,
+            blockTimestamp: BLOCK_TIMESTAMP,
+            borrower: address(0xa11ce),
+            chainId: chainId,
+            collateralAmounts: collateralAmounts,
+            collateralTokenSymbols: collateralTokenSymbols,
+            comet: COMET
+        });
+    }
+
+    function stringArray(string memory string0, string memory string1, string memory string2, string memory string3)
+        internal
+        pure
+        returns (string[] memory)
+    {
+        string[] memory strings = new string[](4);
+        strings[0] = string0;
+        strings[1] = string1;
+        strings[2] = string2;
+        strings[3] = string3;
+        return strings;
+    }
+
+    function uintArray(uint256 uint0, uint256 uint1, uint256 uint2, uint256 uint3)
+        internal
+        pure
+        returns (uint256[] memory)
+    {
+        uint256[] memory uints = new uint256[](4);
+        uints[0] = uint0;
+        uints[1] = uint1;
+        uints[2] = uint2;
+        uints[3] = uint3;
+        return uints;
+    }
+
+    struct ChainPortfolio {
+        uint256 chainId;
+        address account;
+        uint96 nextNonce;
+        string[] assetSymbols;
+        uint256[] assetBalances;
+    }
+
+    function chainAccountsFromChainPortfolios(ChainPortfolio[] memory chainPortfolios)
```

Comment:
We have a `TODO` to refactor `chainAccountsList_` in `QuarkBuilderTest`.

This is one take on an alternate `ChainAccounts[]` constructor.

Reply:
Nice, I like it